### PR TITLE
IDs with the right access can be thrown against doors to open them

### PR DESCRIPTION
**Pull request by DamianX**
:cl:
 * rscadd: IDs and PDAs with the right access can be thrown against doors and airlocks to open them.

### CODE REVIEW

**Comment by MadmanMartian** [comment]
This also makes it so that you can throw *people* with the right access to open doors/airlocks

Can't open a door for fear of shock? Slip the captain and just use him as a battering ram!

**Comment by DamianX** [comment]
That's already a thing, though.

**Comment by bathosbathtime** [comment]
Giving a monkey man your ID and throwing him  during subverted AI times is a high level medbay strat, can confirm.

**Comment by SonixApache** [comment]
Does a similar thing happen for IDs without the proper access?

**Comment by DamianX** [comment]
similar to what?

**Comment by SonixApache** [comment]
Does throwing an ID with not enough access at a door that requires access produce a "You don't have enough access" noise?

**Comment by DamianX** [comment]
Throwing at it anything that's not an ID with access now does that.

**Comment by Dacendeth** [comment]
What about throwing emags at doors or is that too weird, also do wallets work correctly with this?

**Comment by DamianX** [comment]
I'm not gonna do emags with this PR but I have no opinion about it one way or another.
Wallets should work but 0% tested.

**Comment by agentofzog2** [comment]
Seems like it;ll just make it even easier to avoid shocked doors.
Guess I'll just flood plasma like always.

**Comment by jknpj** [comment]
bolt them with the lights disabled, this is still humancreep though

File: code/game/machinery/doors/door.dm [hunk 1]
```diff
@@ -60,6 +60,22 @@ var/list/all_doors = list()
...
+	. = ..()
+	if(.)
+		return
+	var/obj/item/thing = AM
+	if(!istype(thing))
+		return FALSE
+	if(operating || !density)
+		return FALSE
+	set_up_access()
+	if(!can_access(thing.GetAccess(), req_access, req_one_access))
```
Comment (Kurfursten) [inline code/game/machinery/doors/door.dm]: check_access already exists for this function, see access.dm L143

**Review by ShiftyRail — CHANGES_REQUESTED** [review]
Do the check_access thing please

**Review by ShiftyRail — APPROVED** [review]

**Comment by SonixApache** [comment]
🎉

**Pull request merged**